mariner/mariner.go

```diff
@@ -154,8 +152,24 @@ func (c Config) update(version, path string) error {
 
 	return nil
 }
+func (c Config) saveAdvisoryPerYear(dirName string, def Definition) error {
+	// Mariner uses `<ID>_<last_number_from_version>` format for `advisory_id`.
+	// But `advisory_id` is not required field.
+	// Therefore, if `advisory_id` is not exist, we create this field independently.
+	// cf. https://github.com/aquasecurity/vuln-list-update/pull/271#issuecomment-2111678641
+	advisoryID := def.Metadata.AdvisoryID
+	if advisoryID == "" {
+		advisoryID = def.ID
+		// for `0` versions `_0` suffix is omitted.
+		if def.Version != "" && def.Version[len(def.Version)-1:] != "0" {
+			advisoryID = fmt.Sprintf("%s_%s", advisoryID, def.Version[len(def.Version)-1:])
```

Comment:
```suggestion
			advisoryID = fmt.Sprintf("%s-%s", advisoryID, def.Version[len(def.Version)-1:])
```

Reply:
fixed in https://github.com/aquasecurity/vuln-list-update/pull/271/commits/995ea97f301c04bacd37987086c6d8992add1703